Update instruction for naming arguments (#5537)
kubernetes/community: contributors/devel/sig-instrumentation/migration-to-structured-logging.md

argument name and argument value. This means when migrating a log call we need to add an additional string before each
argument, that will be used as it's name.

> How variadic arguments to should be used:

Reviewer: ```suggestion
How variable arguments should be used:
```

}
```

When deciding on names of arguments you should:
* Always use [lowerCamelCase], for example use `containerName` and not `container name` or `container_name`.

> * Use [alphanumeric] characters and not should not any special characters like `%$*`, non-latin or unicode characters.

Reviewer: ```suggestion
* Use [alphanumeric] characters: no special characters like `%$*`, non-latin, or unicode characters.
```

* Use object kind when referencing Kubernetes objects, for example `deployment`, `pod` and `node`.

> * Describe the type of value stored under they key and avoid to:

Reviewer: ```suggestion
* Describe the type of value stored under the key and use normalized labels:
```

>   * Include any of value properties, for example instead of `directory` use `path`.

Reviewer: I don't understand this comment, can you elaborate?

Author: Fact that path points to a directory instead of a file should not influence how it's logged.

Reviewer: ```suggestion
  * Don't include implementation-specific details in the labels. Don't use `directory`, do use `path`.
```

>   * Describe context how value is used, for example instead of `podIP` use `IP`.

Reviewer: ```suggestion
  * Do not provide additional context for how value is used. Don't use `podIP`, do use `IP`.
```

>   * Shorten the name (except acronyms), for example instead of `addr` use `address`

Reviewer: ```suggestion
  * With the exception of acronyms like "IP" and the standard "err", don't shorten names. Don't use `addr`, do use `address`.
```

  * Use ambiguous names, that require falling back on message to undestand what value it is. For example instead of

>     `key` use `cacheKey` or instead of `version` use `dockerVersion`.

Reviewer: This sort of contradicts the guidance above on "don't describe how the value is used", can we make this clearer?

Author: Sorry for not being clear.

Motivation behind this was to avoid overloading some terms that are not really the same thing or cannot be matched or compared. Without this guidance any label key, cache key, annotation key would be put under "key". They are pretty different in what values would you expect and you would never want to compare or match different types.

Similarly to "version", even thou all versions could be using a semantic representation there is no benefit from putting all of them in same key as you would never want to compare  docker version to kubelet version.

I think our overall goal of picking keys is grouping values that we would want to potentially match or compare. Fact that some IP is used by a Pod should not dictate it's name as it would make it hard to correlate all other IP that are not used by Pod. On the other hand "cacheKey" makes sense as you would want to be able to check usage of cache without needing to consider any other type of key.

Please let me know if that make sense as I may be overthinking this.

Reviewer: I wrote a suggested update.

>   * Use ambiguous names, that require falling back on message to undestand what value it is. For example instead of
>     `key` use `cacheKey` or instead of `version` use `dockerVersion`.

Reviewer: ```suggestion
  * When names are very ambiguous, try to include context in the label. For example, instead of
    `key` use `cacheKey` or instead of `version` use `dockerVersion`.
```

* Be consistent, for example when logging file path we should always use `path` and not switch between
  `hostPath`, `path`, `file`.

> Exemptions (some cases are temporary workarounds that may change if we settle on better solution):

Reviewer: ```suggestion
Here are a few exceptions to the rules above---some cases are temporary workarounds that may change if we settle on better solution:
```

> * `err` is one case where short for error should be used, reason is to match key used by `klog.ErrorS`

Reviewer: ```suggestion
* Do use `err` rather than `error` to match the key used by `klog.ErrorS`
```

* Context in name is acceptable to distinguish between values that normally go under same key. For example using both

>   `status` and `oldStatus` in log that needs to show the change between statuses.

Reviewer: Shouldn't the other status be `newStatus` then?

Author: I expect that logs referring to state change will be much rarer than ones describing the state at the moment of writing logs. New state after the change could be considered the current state at the moment of writing logs, so instead of creating new group of keys prefixed with "new", we can use the fact it's the current state for log allowing for correlation with other logs.
